Profile part for the back-end.

alex2011576 opened the pull request.

alex2011576 committed:
Add avatar to users schema

alex2011576 committed:
Server-side type updates

alex2011576 committed:
Updates to  users repository:

-added functionality for user profile data,
- removed all select*

alex2011576 committed:
Router endpoints for user profile.

alex2011576 committed:
Added avatar fetching to auth strategies

alex2011576 committed:
Images validation,parsing, converter.

alex2011576 committed:
profile data parsing and validation

alex2011576 committed:
Prettifier

fglsn commented on server/src/repositories/userRepository.ts:
```diff
@@ -63,8 +73,18 @@ const addNewUser = async (newUser: NewUserWithHashedPwd): Promise<User> => {
+			newUser.firstname,
+			newUser.lastname,
+			newUser.activationCode,
+			true,
```
whats that true thing btw?

fglsn replied:
a wait nevermind, got it :D

fglsn reviewed (approved):
Looks good, nothing to mention

alex2011576 committed:
Merge branch 'master' into aleksei

KaomN reviewed (approved):
Looks good!

alex2011576 committed:
Language in session and user login.
Introduce language in seesion.
Also added language data overall in user flow (e.g. login) with mapper.

alex2011576 committed:
min image reduced to 10x10 px
